docs/proposals/project-user-resource-quotas.md

```diff
@@ -0,0 +1,221 @@
+| KubeVirt     | N                         | We set the requested size directly into MachineDeployment |                                  |
+| Nutanix      | TBD                       | TBD                                                    |                                  |
+| Equinox      | Y                         |                                                        |                                  |
+| vSphere      | N                         | We set the requested size directly into MachineDeployment |                                  |
+
+
+### Enforcing Quotas - Implementation variants
+
+I see 2 possible ways to implement the enforcing of project quotas through MachineDeployments. 
+
+One is through a MachineDeployment webhook, although it requires moving the MD's to the seed cluster, because the webhook 
+needs to be out of reach for the user-cluster users. The pros of that is that users can get directly notified if they 
+can't create a MachineDeployment, but the con is that we would need to do a migration and change the shared code with KubeOne.
+
+In the other approach, the machine-controller would be responsible for checking the quota before creating nodes for a MachineDeployment.
+The pro is that it wouldn't be a big change, the con is that it won't be so visible to users, we could get questions "why are my nodes not being created".
+We can show the info that there is not enough resource quota left to create the nodes, but users would only see that in MachineDeployment
+Events. So it would be something like Pods not getting scheduled because there is no space.
+
+
+#### 1. Webhook for MachineDeployments
+
+This variant is dependent on migrating the MachineDeployments from being user-cluster resources to being seed-cluster resources.
+
+If the MD's are on the seed cluster, we can control their creation/modification through a validation webhook which would check with the 
+project's resource quota if the MD fits. If not, we would just deny the request.
+
+The problem here is if its possible and at what cost to migrate the MD's, while thinking about KubeOne as well.
+We could eventually dynamically set the kubeconfig of the cluster where the machine-controller watches for MachineDeployments, 
+and set it to the seed-cluster in case of KKP. But then, as MachineDeployments are cluster-scoped we will need additional 
+cluster info label or something on them.
+
+Another issue could be that cluster users won't have direct kubectl access to MD's anymore, at least until we start 
+supporting it somewhere in the future. But they can use the KKP API. 
+
+#### 2. Machine-controller checks quota
+
+In this variant, the machine-controller is checking the quota before it creates nodes for a MachineDeployment. 
+If there is no quota left, it should not schedule the node, and emit some event and set status that there is not enough resources
+left in the quota.
+
+### How to ensure that we get proper resource usage and that it stays in sync?
```

Comment:
There is a section at line 93 with the same title. I assume the two should be merged in some way.